qchat/logic/slash_commands.py

```diff
@@ -0,0 +1,243 @@
+            )
+
+    def cmd_shrug(self, args: str) -> SlashCommandResult:
+        """Shrug command: ¯\\_(ツ)_/¯
+
+        :param args: optional message after shrug
+        :return: SlashCommandResult
+        """
+        message = "¯\\_(ツ)_/¯"
+        if args:
+            message = f"{message} {args}"
+        return SlashCommandResult(success=True, message_text=message)
+
+    def cmd_tableflip(self, args: str) -> SlashCommandResult:
+        """Table flip command: (╯°□°)╯︵ ┻━┻
+
+        :param args: optional message after tableflip
+        :return: SlashCommandResult
+        """
+        message = "(╯°□°)╯︵ ┻━┻"
+        if args:
+            message = f"{message} {args}"
+        return SlashCommandResult(success=True, message_text=message)
+
+    def cmd_lenny(self, args: str) -> SlashCommandResult:
+        """Lenny face command: ( ͡° ͜ʖ ͡°)
+
+        :param args: optional message after lenny
+        :return: SlashCommandResult
+        """
+        message = "( ͡° ͜ʖ ͡°)"
+        if args:
+            message = f"{message} {args}"
+        return SlashCommandResult(success=True, message_text=message)
+
+    def cmd_yolo(self, args: str) -> SlashCommandResult:
+        """YOLO command: converts message to CAPS + emoji
+
+        :param args: message to yolo-fy
+        :return: SlashCommandResult
+        """
+        if not args:
+            message = "YOLO! 🎲"
+        else:
+            message = f"{args.upper()} 🎲"
+        return SlashCommandResult(success=True, message_text=message)
+
+    def cmd_flip(self, args: str) -> SlashCommandResult:
+        """Coin flip command: heads or tails
+
+        :param args: unused
+        :return: SlashCommandResult
+        """
+        result = random.choice(["Heads", "Tails"])
+        emoji = "🪙"
+        message = f"{emoji} {result}!"
+        return SlashCommandResult(success=True, message_text=message)
+
+    def cmd_roll(self, args: str) -> SlashCommandResult:
+        """Dice roll command: 1-6
+
+        :param args: optional dice specification (e.g., "2d20" for 2 dice of 20 sides)
+        :return: SlashCommandResult
+        """
+        # Parse args for custom dice (e.g., "2d20")
+        num_dice = 1
+        num_sides = 6
+
+        if args:
+            try:
+                if "d" in args.lower():
+                    parts = args.lower().split("d")
+                    num_dice = int(parts[0]) if parts[0] else 1
+                    num_sides = int(parts[1]) if parts[1] else 6
+                else:
+                    num_sides = int(args)
+            except (ValueError, IndexError):
+                return SlashCommandResult(
+                    success=False,
+                    error="Invalid format. Use /roll [faces] or /roll [number]d[faces]",
+                )
+
+        # Validate
+        if num_dice < 1 or num_dice > 100:
+            return SlashCommandResult(
+                success=False, error="Invalid number of dice (1-100)"
```

Comment:
```suggestion
                success=False, error="Invalid number of dice [1-100]"
```

Or a more verbose error like `"Invalid number of dice (must be between 1 and 100)"`, but maybe that's too long...